baseapp/router.go

```diff
@@ -34,22 +37,44 @@ func NewRouter() *router {
 var isAlpha = regexp.MustCompile(`^[a-zA-Z]+$`).MatchString
 
 // AddRoute - TODO add description
-func (rtr *router) AddRoute(r string, h sdk.Handler) Router {
+func (rtr *router) AddRoute(r string, h sdk.Handler, i sdk.InitGenesis) Router {
 	if !isAlpha(r) {
 		panic("route expressions can only contain alphanumeric characters")
 	}
-	rtr.routes = append(rtr.routes, route{r, h})
+	rtr.routes = append(rtr.routes, route{r, h, i})
 
 	return rtr
 }
 
-// Route - TODO add description
 // TODO handle expressive matches.
+func matchRoute(path string, route string) bool {
+	return path == route
+}
+
+// Route - TODO add description
 func (rtr *router) Route(path string) (h sdk.Handler) {
 	for _, route := range rtr.routes {
-		if route.r == path {
+		if matchRoute(path, route.r) {
 			return route.h
 		}
 	}
 	return nil
 }
+
+func (rtr *router) RouteGenesis(path string) (i sdk.InitGenesis) {
+	for _, route := range rtr.routes {
+		if matchRoute(path, route.r) {
+			return route.i
+		}
+	}
+	return nil
+}
+
+func (rtr *router) ForEach(f func(string, sdk.Handler, sdk.InitGenesis) error) error {
```

Comment:
Need a comment, enable go-metalinter

Reply:
Maybe a different name than ForEach - or better explanation

Reply:
Instead of `ForEach` let's create something called `InitModules` which performs the loop over all the modules.... if this is possible